interval.hh

```diff
@@ -730,6 +730,15 @@ public:
     friend std::ostream& operator<<(std::ostream& out, const nonwrapping_interval<U>& r);
 };
 
+template <typename T>
+nonwrapping_interval<T> reversed(nonwrapping_interval<T> i) {
```

Comment:
Please add a comment that explains that the reversed interval must be used with a comparator that has the reverse sense of the one that was used to interpret the input.